packages/vue-treant/src/Treant.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="treant" :style="{ 'max-width': width }">
+    <div class="tree-container"></div>
+  </div>
+</template>
+
+<script>
+  import 'treantjs';
+  import 'treantjs/Treant.css';
+  import Raphael from 'treantjs/vendor/raphael';
+  window.Raphael = Raphael;
+
+  export default {
+    props: ['tree', 'highlightedNode', 'config'],
+    data() {
+      return {
+        treant: null,
+        width: '600px',
+      };
+    },
+    methods: {
+      onNodeEnter(e) {
+        const { text } = e.target.data.treenode;
+        this.$emit('enter', text);
+      },
+      onNodeLeave(e) {
+        this.$emit('leave');
+      },
+      onTreeLoaded(args) {
+        const nodes = this.$el.querySelectorAll('.node');
+        nodes.forEach(node => {
+          node.addEventListener('mouseenter', this.onNodeEnter);
+          node.addEventListener('mouseleave', this.onNodeLeave);
+        });
+      },
+      setWidth() {
+        this.width = `${this.$parent.$el.clientWidth}px`;
+      },
+      onResize() {
```

Comment:
What about attaching a watcher to sidebar toggles?

e.g, if the user toggles a sidebar on / off, force redraw?

Reply:
Yeah, I think we need to do that in addition to what I'm doing here.